emcc.py

```diff
@@ -1944,15 +1952,22 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
         combined.close()
       # finish compiling to WebAssembly, using asm2wasm, if we didn't already emit WebAssembly directly using the wasm backend.
       if not shared.Settings.WASM_BACKEND:
-        cmd = [os.path.join(binaryen_bin, 'asm2wasm'), asm_target, '--mapped-globals=' + wasm_text_target + '.mappedGlobals', '--total-memory=' + str(shared.Settings.TOTAL_MEMORY)]
+        cmd = [os.path.join(binaryen_bin, 'asm2wasm'), asm_target, '--total-memory=' + str(shared.Settings.TOTAL_MEMORY)]
         if shared.Settings.BINARYEN_IMPRECISE:
           cmd += ['--imprecise']
         if opt_level == 0:
           cmd += ['--no-opts']
+        # import mem init file if it exists, and if we will not be using asm.js as a binaryen method (as it needs the mem init file, of course)
+        import_mem_init = memory_init_file and 'asmjs' not in shared.Settings.BINARYEN_METHOD and 'interpret-asm2wasm' not in shared.Settings.BINARYEN_METHOD
+        if import_mem_init and os.path.exists(memfile):
+          cmd += ['--mem-init=' + memfile]
         logging.debug('asm2wasm (asm.js => WebAssembly): ' + ' '.join(cmd))
         TimeLogger.update()
         subprocess.check_call(cmd, stdout=open(wasm_text_target, 'w'))
         log_time('asm2wasm')
+        if import_mem_init:
+          shared.try_delete(memfile)
```

Comment:
This unconditional delete feels odd - should this distinguish between if the user wanted mem init, or if we internally enabled mem init, and only delete if mem init was enabled internally and not by the user? I'm thinking about the scenario where one is interested in serving both asm.js and wasm content, with asm.js being the fallback (and having mem-init file)